api/src/main/java/org/openmrs/module/dhisconnector/api/util/DHISConnectorUtil.java

```diff
@@ -0,0 +1,51 @@
+package org.openmrs.module.dhisconnector.api.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class DHISConnectorUtil {
+
+    private static final Logger log = LoggerFactory.getLogger(DHISConnectorUtil.class);
+
+    public static String zipMappingBundle (String tempDirectory, List<File> fileList) throws IOException {
+        String path = tempDirectory + "mappingBundle_" + (new Date()).getTime() + ".zip";
+        FileOutputStream fOut = new FileOutputStream(path);
+        ZipOutputStream zOut = new ZipOutputStream(fOut);
+        for (File file: fileList) {
+            FileInputStream fis = new FileInputStream(file);
+            ZipEntry zipEntry = new ZipEntry(file.getName());
+            zOut.putNextEntry(zipEntry);
+
+            byte[] bytes = new byte[1024];
+            int length;
+            while ((length = fis.read(bytes)) >= 0) {
+                zOut.write(bytes, 0 , length);
+            }
+            fis.close();
+        }
+        zOut.close();
```

Comment:
Are these streams closed when an exception is thrown?

Reply:
@dkayiwa 
Oh yeah it slipped my mind, I'll use a try, catch, finally block to close the stream, thank you